src/main/java/de/tum/in/www1/artemis/web/rest/ComplaintResource.java

```diff
@@ -463,8 +486,9 @@ private void filterOutUselessDataFromComplaints(List<Complaint> complaints, bool
         complaints.forEach(complaint -> {
             String submissorName = principal.getName();
             User assessor = complaint.getResult().getAssessor();
+            User student = complaint.getStudent();
 
-            if (assessor.getLogin().equals(submissorName) == assessorSameAsCaller) {
+            if (assessor.getLogin().equals(submissorName) == assessorSameAsCaller && (student != null && assessor.getLogin().equals(student.getLogin())) == testRun) {
```

Comment:
Could it happen for normal exercises that (student != null && assessor.getLogin().equals(student.getLogin())) evaluates to true because the assessor was the same person (e.g. for a quick testing session) and testRun is false? Might not matter in this case but I'm curios.

Reply:
`(student != null && assessor.getLogin().equals(student.getLogin()))` yes as you can assess your own submissions. But for normal (not test run participations) `assessorSameAsCaller` is set to false 
which would in turn make `assessor.getLogin().equals(submissorName) == assessorSameAsCaller` result to false thereby falsifying the clause. There should therefore be no interference here between test run participations and normal participations